diff --git a/crates/atuin/src/shell/atuin.zsh b/crates/atuin/src/shell/atuin.zsh
index d580f704..b4de7b2c 100644
--- a/crates/atuin/src/shell/atuin.zsh
+++ b/crates/atuin/src/shell/atuin.zsh
@@ -56,7 +56,20 @@ _atuin_search() {
     # TODO: not this
     local output
     # shellcheck disable=SC2048
-    output=$(ATUIN_SHELL_ZSH=t ATUIN_LOG=error ATUIN_QUERY=$BUFFER atuin search $* -i 3>&1 1>&2 2>&3)
+    if [ -z "${TMUX}" ]; then
+        output=$(ATUIN_SHELL_ZSH=t ATUIN_LOG=error ATUIN_QUERY=$BUFFER atuin search $* -i -- 3>&1 1>&2 2>&3)
+    else
+        tmpdir=$(mktemp -d)
+        mkfifo "$tmpdir/pipe"
+        trap "rm -rf '$tmpdir'" EXIT HUP INT TERM
+        output=$(
+            cat "$tmpdir/pipe" &;
+            tmux display-popup -d $(pwd) -B -E -E -h 80% -w 80% -- \
+                "$(printf "%q " ATUIN_SHELL_ZSH=t ATUIN_LOG=error ATUIN_QUERY=$BUFFER atuin search $* -i --) 1>&2 2>$tmpdir/pipe"
+        )
+        rm -rf "$tmpdir"
+    fi
+
 
     zle reset-prompt
 
