diff --git a/src/Language/Haskell/Brittany/Internal/Layouters/Module.hs b/src/Language/Haskell/Brittany/Internal/Layouters/Module.hs
index 4416141..1acd070 100644
--- a/src/Language/Haskell/Brittany/Internal/Layouters/Module.hs
+++ b/src/Language/Haskell/Brittany/Internal/Layouters/Module.hs
@@ -64,6 +64,7 @@ layoutModule lmod@(L _ mod') = case mod' of
                 , docWrapNode lmod $ appSep $ case les of
                   Nothing -> docEmpty
                   Just x  -> layoutLLIEs True KeepItemsUnsorted x
+                , docSeparator
                 , docLit $ Text.pack "where"
                 ]
             addAlternative
@@ -71,11 +72,13 @@ layoutModule lmod@(L _ mod') = case mod' of
               [ docAddBaseY BrIndentRegular $ docPar
                 (docSeq [appSep $ docLit $ Text.pack "module", docLit tn]
                 )
-                (docWrapNode lmod $ case les of
-                  Nothing -> docEmpty
-                  Just x  -> layoutLLIEs False KeepItemsUnsorted x
+                (docSeq [ docWrapNode lmod $ case les of
+                            Nothing -> docEmpty
+                            Just x  -> layoutLLIEs False KeepItemsUnsorted x
+                        , docSeparator
+                        , docLit $ Text.pack "where"
+                        ]
                 )
-              , docLit $ Text.pack "where"
               ]
           ]
       : (commentedImportsToDoc <$> sortCommentedImports commentedImports) -- [layoutImport y i | (y, i) <- sortedImports]
