diff --git a/config/nixpkgs/home.nix b/config/nixpkgs/home.nix
index 55c3a7c..07c9f44 100644
--- a/config/nixpkgs/home.nix
+++ b/config/nixpkgs/home.nix
@@ -30,7 +30,9 @@ in {
     ./home/git.nix
     ./home/tmux.nix
     ./home/neovim.nix
-    ./home/pc.nix
+    # ./home/pc.nix
+    ./home/haskell.nix
+    ./home/coc-nvim.nix
   ] ++ lib.optional (builtins.getEnv "BANKID" != "") ./home/scb.nix;
 
   home.username = "j";
diff --git a/config/nixpkgs/home/coc-nvim.nix b/config/nixpkgs/home/coc-nvim.nix
index d796830..a9721f7 100644
--- a/config/nixpkgs/home/coc-nvim.nix
+++ b/config/nixpkgs/home/coc-nvim.nix
@@ -248,7 +248,7 @@
       };
       languageserver = {
         clangd = {
-          command = "${pkgs.clang-tools}/bin/clangd";
+          # command = "${pkgs.clang-tools}/bin/clangd";
           args = [ "--background-index" "--compile-commands-dir=build" ];
           rootPatterns = [
             "compile_flags.txt"
diff --git a/config/nixpkgs/home/haskell.nix b/config/nixpkgs/home/haskell.nix
index 42740f9..75e3dab 100644
--- a/config/nixpkgs/home/haskell.nix
+++ b/config/nixpkgs/home/haskell.nix
@@ -139,7 +139,7 @@ in {
       ];
 
       haskell-language-server = super.haskell-language-server.override {
-        supportedGhcVersions = [ "884" "8102" ];
+        supportedGhcVersions = [ "8102" ];
       };
 
       docServer = self.writeShellScriptBin "doc-server" ''
