diff --git a/autoload/neomake.vim b/autoload/neomake.vim
index 323d3e9..37d4610 100644
--- a/autoload/neomake.vim
+++ b/autoload/neomake.vim
@@ -1384,13 +1384,11 @@ function! s:AddExprCallback(jobinfo, prev_list) abort
                 call remove(list, k)
             endfor
         endif
-        if file_mode
-            call setloclist(0, list, 'r')
-        else
-            call setqflist(list, 'r')
-        endif
+        call setqflist(list, 'r')
     endif
 
+    call lightline#update()
+
     if !empty(bufnr_from_temp) || !empty(bufnr_from_stdin)
         if !has_key(make_info, '_wipe_unlisted_buffers')
             let make_info._wipe_unlisted_buffers = []
