diff --git a/autoload/hdevtools.vim b/autoload/hdevtools.vim
index 73fd9a2..7d955fe 100644
--- a/autoload/hdevtools.vim
+++ b/autoload/hdevtools.vim
@@ -102,7 +102,7 @@ function! hdevtools#info(identifier)
     return
   endif
 
-  let l:file = expand('%')
+  let l:file = expand('%:p')
   if l:file ==# ''
     call hdevtools#print_warning("current version of hdevtools.vim doesn't support running on an unnamed buffer.")
     return
@@ -333,6 +333,7 @@ function! s:infowin_create()
   exe 'silent! botright 1split' fnameescape("(Info)")
   setlocal noinsertmode     " don't make Insert mode the default
   setlocal report=9999      " don't show 'X lines changed' reports
+  setlocal scrolloff=0      " no scrolloff
   setlocal sidescroll=0     " don't sidescroll in jumps
   setlocal sidescrolloff=0  " don't sidescroll automatically
   setlocal noequalalways    " don't auto-balance window sizes
@@ -373,7 +374,7 @@ function! s:infowin_create()
   highlight link HdevtoolsInfoLink Underlined
 endfunction
 
-function! s:infowin_close()
+function! hdevtools#infowin_close()
   exe "silent! bunload!" s:hdevtools_info_buffer
   let s:hdevtools_info_buffer = -1
 endfunction
@@ -510,7 +511,7 @@ function! hdevtools#type()
     return b:hdevtools_type.type()
   endif
 
-  let l:file = expand('%')
+  let l:file = expand('%:p')
   if l:file ==# ''
     call hdevtools#print_warning("current version of hdevtools.vim doesn't support running on an unnamed buffer.")
     return ['', '']
diff --git a/ftplugin/haskell/hdevtools.vim b/ftplugin/haskell/hdevtools.vim
index dc19531..789e579 100644
--- a/ftplugin/haskell/hdevtools.vim
+++ b/ftplugin/haskell/hdevtools.vim
@@ -6,7 +6,7 @@ let b:did_ftplugin_hdevtools = 1
 if !exists('s:has_hdevtools')
   let s:has_hdevtools = 0
 
-  if !executable('stack exec -- hdevtools')
+  if !executable('hdevtools')
     call hdevtools#print_error('hdevtools: hdevtools is not executable!')
     finish
   endif
