diff --git a/autoload/LanguageClient.vim b/autoload/LanguageClient.vim
index 992ff0e..c844fca 100644
--- a/autoload/LanguageClient.vim
+++ b/autoload/LanguageClient.vim
@@ -210,9 +210,6 @@ function! s:FZF(source, sink) abort
                 \ 'sink': function(a:sink),
                 \ 'options': l:options,
                 \ }))
-    if has('nvim')
-        call feedkeys('i')
-    endif
 endfunction
 
 function! s:Edit(action, path) abort
