diff --git a/autoload/neomake/makers/ft/c.vim b/autoload/neomake/makers/ft/c.vim
index b104580..6cf94e1 100644
--- a/autoload/neomake/makers/ft/c.vim
+++ b/autoload/neomake/makers/ft/c.vim
@@ -13,6 +13,7 @@ function! neomake#makers#ft#c#clang() abort
     " as a single-file maker, include the current directory in the default
     " search path
     return {
+        \ 'exe': 'clang',
         \ 'args': ['-fsyntax-only', '-Wall', '-Wextra', '-I./'],
         \ 'errorformat':
             \ '%-G%f:%s:,' .
