diff --git a/autoload/neomake/makers/ft/haskell.vim b/autoload/neomake/makers/ft/haskell.vim
index 4dcd962..822e47b 100644
--- a/autoload/neomake/makers/ft/haskell.vim
+++ b/autoload/neomake/makers/ft/haskell.vim
@@ -142,6 +142,20 @@ function! neomake#makers#ft#haskell#liquid() abort
       \ })
 endfunction
 
+function! neomake#makers#ft#haskell#mu() abort
+    return {
+        \ 'exe': 'mu',
+        \ 'args': ["Strat.IDE.TypeCheck"],
+        \ 'errorformat':
+            \ '%EERROR %f %l:%c: %m,' .
+            \ '%EERROR %f %l:%c:,' .
+            \ '%WWARNING %f %l:%c: %m,' .
+            \ '%WWARNING %f %l:%c:,' .
+            \ '%WWARNING %f %l:%c: ,' .
+            \ '%+C%.%#'
+        \ }
+endfunction
+
 " @vimlint(EVL103, 1, a.job_id)
 " @vimlint(EVL103, 1, a.event)
 " @vimlint(EVL101, 1, l.self)
