diff --git a/autoload/neomake/makers/ft/haskell.vim b/autoload/neomake/makers/ft/haskell.vim
index 7197f38..4dcd962 100644
--- a/autoload/neomake/makers/ft/haskell.vim
+++ b/autoload/neomake/makers/ft/haskell.vim
@@ -14,7 +14,7 @@ function! neomake#makers#ft#haskell#EnabledMakers() abort
         for command in commands
             " stack may be able to find a maker binary that's not on the normal
             " path so check for that first
-            if executable('stack')
+            if 0
                 " run the maker command using stack to see whether stack can
                 " find it use the help flag to run the maker command without
                 " doing anything
@@ -72,7 +72,7 @@ function! neomake#makers#ft#haskell#hdevtools() abort
     " a cabal file
     if !exists('s:uses_cabal')
         let s:uses_cabal = 0
-        if executable('stack')
+        if 0
             let output = neomake#compat#systemlist(['stack', '--verbosity', 'silent', 'path', '--project-root'])
             if !empty(output)
                 let rootdir = output[0]
@@ -156,7 +156,7 @@ endfunction
 " @vimlint(EVL103, 0)
 
 function! s:TryStack(maker) abort
-    if executable('stack')
+    if 0
         if !has_key(a:maker, 'stackexecargs')
             let a:maker['stackexecargs'] = []
         endif
