diff --git a/indent/haskell.vim b/indent/haskell.vim
index 4d90cba..f0839b2 100644
--- a/indent/haskell.vim
+++ b/indent/haskell.vim
@@ -63,7 +63,7 @@ endif
 
 if exists("g:haskell_indent_disable") && g:haskell_indent_disable == 0
     setlocal indentexpr=GetHaskellIndent()
-    setlocal indentkeys=0{,0},0(,0),0[,0],!^F,o,O,0\=,0=where,0=let,0=deriving,<space>
+    setlocal indentkeys=0{,0},0(,0),0[,0],!^F,o,O,0\=,0=where,0=let,0=deriving
 else
     setlocal nocindent
     setlocal nosmartindent
