diff --git a/autoload/necoghc.vim b/autoload/necoghc.vim
index 47a20ee..f8bde53 100644
--- a/autoload/necoghc.vim
+++ b/autoload/necoghc.vim
@@ -109,7 +109,7 @@ function! s:to_desc(sym, dict) abort
   if has_key(a:dict, 'kind')
     let l:desc .= printf('%s %s %s', a:dict.kind, a:sym, a:dict.args)
   elseif has_key(a:dict, 'type')
-    let l:desc .= printf('%s :: %s', a:sym, a:dict.type)
+    let l:desc .= printf(': %s', a:dict.type)
   else
     let l:desc .= a:sym
   endif
diff --git a/rplugin/python3/deoplete/sources/ghc.py b/rplugin/python3/deoplete/sources/ghc.py
index a914333..015d0e3 100644
--- a/rplugin/python3/deoplete/sources/ghc.py
+++ b/rplugin/python3/deoplete/sources/ghc.py
@@ -12,7 +12,7 @@ class Source(Base):
         Base.__init__(self, vim)
 
         self.name = 'ghc'
-        self.mark = '[ghc]'
+        self.mark = '[λ]'
         self.filetypes = ['haskell', 'lhaskell']
         self.is_bytepos = True
         self.rank = 500
