diff --git a/python/haskell-textobj.py b/python/haskell-textobj.py
index ad6457c..4a7fa1b 100755
--- a/python/haskell-textobj.py
+++ b/python/haskell-textobj.py
@@ -119,7 +119,10 @@ def extend_typesig(start_line, end_line, lines):
     while is_comment(lines[first_line]):
         first_line += 1
 
-    if "::" in lines[first_line].split()[1]:
+    first_line_split = lines[first_line].split()
+    if len(first_line_split) > 1 and "::" in first_line_split[1]:
+        return start2, end_line
+    elif len(first_line_split) == 1 and "::" in lines[first_line+1].split():
         return start2, end_line
     else:
         return start_line, end_line
